packages/electron-updater/src/providers/GitHubProvider.ts

```diff
@@ -77,7 +106,7 @@ export class GitHubProvider extends BaseGitHubProvider<GithubUpdateInfo> {
       throw newError(`No published versions on GitHub`, "ERR_UPDATER_NO_PUBLISHED_VERSIONS")
     }
 
-    const channelFile = getChannelFilename(this.getDefaultChannelName())
+    const channelFile = getChannelFilename(this.updater.allowPrerelease ? this.getCustomChannelName(String(semver.prerelease(tag)?.[0]) || 'latest') : this.getDefaultChannelName())
```

Comment:
`String(undefined) || 'latest'` will always return `undefined`
I think you meant to do:
```
String(semver.prerelease(tag)?.[0] || 'latest')
```

Reply:
Thanks, changed.

Reply:
This still isn't valid:
`String(semver.prerelease(tag)?.[0]) || 'latest'`

It needs to be 
`String(semver.prerelease(tag)?.[0] || 'latest')`